File: e2e-tests/spring-rest-opensearch/src/main/java/com/opensearch/config/SwaggerConfiguration.java
package com.opensearch.config;

import static springfox.documentation.builders.PathSelectors.regex;

import org.springframework.context.annotation.Bean;
import org.springframework.security.core.Authentication;
import org.springframework.web.context.request.WebRequest;
import springfox.documentation.builders.ApiInfoBuilder;
import springfox.documentation.service.ApiInfo;
import springfox.documentation.spi.DocumentationType;
import springfox.documentation.spring.web.plugins.Docket;

public class SwaggerConfiguration {

    private final String path;

    public SwaggerConfiguration(String path) {
        this.path = path;
    }

    @Bean
    public Docket docketApi() {
        return new Docket(DocumentationType.SWAGGER_2)
                .apiInfo(apiInfo())
                .select()
                .paths(regex("/" + path + "/.*"))
                .build()
                .ignoredParameterTypes(WebRequest.class, Authentication.class);
    }

    private ApiInfo apiInfo() {
        return new ApiInfoBuilder()
                .title("OpenSearch Sample API")
                .description("OpenSearch Sample API description for " + path)
                .version("1.0")
                .build();
    }
}

File: core-tests/e2e-tests/spring/spring-rest-multidb-mongo-redis-postgres/src/main/kotlin/com/foo/spring/rest/multidb/mongoredispostgres/SwaggerConfiguration.kt
package com.foo.spring.rest.multidb.mongoredispostgres

import org.springframework.context.annotation.Bean
import org.springframework.security.core.Authentication
import org.springframework.web.context.request.WebRequest
import springfox.documentation.builders.ApiInfoBuilder
import springfox.documentation.service.ApiInfo
import springfox.documentation.spi.DocumentationType
import springfox.documentation.spring.web.plugins.Docket
import springfox.documentation.builders.PathSelectors.regex

open class SwaggerConfiguration {

    @Bean
    open fun docketApi(): Docket {
        return Docket(DocumentationType.SWAGGER_2)
            .apiInfo(apiInfo())
            .select()
            .paths(regex("/api/.*"))
            .build()
            .ignoredParameterTypes(WebRequest::class.java, Authentication::class.java)
    }

    private fun apiInfo(): ApiInfo {
        return ApiInfoBuilder()
            .title("Multi-DB Mongo Redis Postgres API")
            .description("API with endpoints reading Mongo, Redis and Postgres")
            .version("1.0")
            .build()
    }
}
